src/libraries/System.Private.Xml/src/System/Xml/XPath/XPathNavigator.cs

```diff
@@ -1241,7 +1243,7 @@ public virtual object Evaluate(XPathExpression expr, XPathNodeIterator context)
 
         public virtual bool Matches(XPathExpression expr)
         {
-            CompiledXpathExpr cexpr = expr as CompiledXpathExpr;
+            CompiledXpathExpr? cexpr = expr as CompiledXpathExpr;
```

Comment:
please verify `prefix` and `namespaceURI` are non-nullable. I.e. `CreateAttribute` seems like it could be nullable

Reply:
Same as https://github.com/dotnet/runtime/pull/39821#discussion_r459759685.

Properties `Perfix` and `NamespaceURI` are also fine being not-null.